src/s3ql/http.py

```diff
@@ -0,0 +1,1558 @@
+                    await PollNeeded(self._sock.fileno(), POLLIN)
+                elif res == 0:
+                    raise ConnectionClosed('server closed connection')
+                else:
+                    break
+
+        log.debug('found substr at %d', idx)
+        idx += len(substr)
+        buf = rbuf.d[rbuf.b : idx]
+        rbuf.b = idx
+
+        if parts:
+            parts.append(buf)
+            buf = b''.join(parts)
+
+        try:
+            return buf.decode('latin1')
+        except UnicodeDecodeError:
+            raise InvalidResponse('server response cannot be decoded to latin1')
+
+    def _try_fill_buffer(self):
+        '''Try to fill up read buffer
+
+        Returns the number of bytes read into buffer, or `None` if no data was
+        available on the socket. Return zero if the TCP connection has been
+        properly closed but the socket object still exists. On other problems
+        (e.g. if the socket object has been destroyed or the connection
+        interrupted), raise `ConnectionClosed`.
+        '''
+
+        log.debug('start')
+        rbuf = self._rbuf
+
+        # If buffer is empty, reset so that we start filling from beginning
+        if rbuf.b == rbuf.e:
+            rbuf.b = 0
+            rbuf.e = 0
+
+        # There should be free capacity
+        assert rbuf.e < len(rbuf.d)
+
+        if self._sock is None:
+            raise ConnectionClosed('connection has been closed locally')
+
+        try:
+            len_ = self._sock.recv_into(memoryview(rbuf.d)[rbuf.e :])
+            if self.trace_fh:
+                self.trace_fh.write(rbuf.d[rbuf.e : rbuf.e + len_])
+        except (socket.timeout, ssl.SSLWantReadError, BlockingIOError):
+            log.debug('done (nothing ready)')
+            return None
+        except (ConnectionResetError, BrokenPipeError, ssl.SSLEOFError):
+            raise ConnectionClosed('connection was interrupted')
+
+        rbuf.e += len_
+        log.debug('done (got %d bytes)', len_)
+        return len_
+
+    async def _co_fill_buffer(self, len_):
+        '''Make sure that there are at least *len_* bytes in buffer'''
+
+        rbuf = self._rbuf
+        if len_ > len(rbuf.d):
+            raise ValueError('Requested more bytes than buffer has capacity')
+        while len(rbuf) < len_:
+            if len(rbuf.d) - rbuf.b < len_:
+                self._rbuf.compact()
+            res = self._try_fill_buffer()
+            if res is None:
+                await PollNeeded(self._sock.fileno(), POLLIN)
+            elif res == 0:
+                raise ConnectionClosed('server closed connection')
+
+    def readall(self):
+        '''placeholder, will be replaced dynamically'''
+        return eval_coroutine(self.co_readall(), self.timeout)
+
+    async def co_readall(self):
+        '''Read and return complete response body'''
+
+        if self._in_remaining is None:
+            return b''
+
+        log.debug('start')
```

Comment:
A more descriptive log entry would make it much easier to identify this entry and what it means in a debug log file.

Reply:
I'm not sure what else might be relevant her? Note that the debug log output includes the function name automatically.